src/app/vertextool/qgsvertextool.cpp

```diff
@@ -2802,3 +2802,11 @@ void QgsVertexTool::cleanEditor( QgsFeatureId id )
     cleanupVertexEditor();
   };
 }
+
+void QgsVertexTool::clean()
+{
+  if ( mDraggingVertex )
```

Comment:
Should also check against `mDraggingEdge`. There is also a third case of a rubberband, that of the vertex selection rectangle.

Reply:
`mDraggingEdge` --> right.

On the other hand, `cadCanvasReleaseEvent()` calls `stopSelectionRubberBand()`. So, the selection rectangle rubber band is being correctly removed. I'd say we don't need to check for that one, but let me know if (and why) you consider we should.

Reply:
There's an edge case, if one stops editing by using a key bind while still dragging for a selection with the mouse. It's not easy to happen, but a `clean()` method should clean everything.

Reply:
Just replicated it. Agree, it should be removed as well, even if it's an edge case.

Reply:
Solved in https://github.com/qgis/QGIS/pull/36301/commits/1c55cb9ab03579a0879a43d5fba8b1d30356ccfe